source/agora/node/Validator.d

```diff
@@ -83,6 +85,9 @@ public class Validator : FullNode, API
             this.network.discover(required_peer_keys);
             this.network.startPeriodicCatchup(this.ledger, &this.nominator.isNominating);
         });
+
+        // call checkRevealPreimage periodically every 10 seconds
+        this.taskman.timer(10.seconds, &this.checkRevealPreimage, Periodic.Yes);
```

Comment:
I think this routine should not be here. It should be after `this.network.discover`. Otherwise we will try to send preimages before the network has been discovered, which isn't really useful at that point. There will be no nodes to send the preimages to.

Reply:
Note that it should be completely fine to start a timer within a task.

Reply:
I agree with @AndrejMitrovic 
I don't think there's a suitable call point yet.
Timer should only be called once in the node life cycle.

Oh, does our node have a restart function after starting?

Reply:
I mean, you can move it below

```
this.network.discover(required_peer_keys);
>here<
```

That should work fine?

Reply:
I misunderstood.  work fine :)